releng/jenkins/bot-updates/Jenkinsfile

```diff
@@ -325,6 +328,69 @@ pipeline {
       } // steps
     } // stage 'Update Tycho Version'
 
+    stage('Update MWE Version') {
+      when {
+        expression { params.UPDATE_TYPE == 'MWE_VERSION' }
+      }
+      steps {
+        script {
+          // set pull request title message
+          env.MWE1_VERSION="${params.UPDATE_VALUE}".split(',')[0]
+          env.MWE2_VERSION="${params.UPDATE_VALUE}".split(',')[1]
+          env.MWE_P2_URL="${params.UPDATE_VALUE}".split(',')[2]
+          env.PR_TITLE="[releng] Update MWE to ${env.MWE1_VERSION}/${env.MWE2_VERSION}"
+
+          currentBuild.displayName = "#${env.BUILD_NUMBER} (MWE ${env.MWE2_VERSION})";
+          REPOSITORY_NAMES.split(',').each {
+            dir (it) {
+              sh """
+                for f in \$(find . -name *.target -type f); 
+                do
+                  sed -i -e 's|https://.*/emft/mwe/updates/[^"]*|${env.MWE_P2_URL}|' \$f
+                done
+              """
+            } // dir
+          } // for each repository
+          if (fileExists('xtext-lib')) { dir('xtext-lib') {
+            sh """
+              sed -i -E 's|(org\\.eclipse\\.emf\\.mwe\\..*):[^"]*|\\1:${env.MWE1_VERSION}|g' org.eclipse.xtext.dev-bom/build.gradle
+              sed -i -E 's|(org\\.eclipse\\.emf\\.mwe2\\..*):[^"]*|\\1:${env.MWE2_VERSION}|g' org.eclipse.xtext.dev-bom/build.gradle
+            """
+          }}
+          if (fileExists('xtext-core')) { dir('xtext-core') {
+            sh """
+              for f in \$(find org.eclipse.xtext.tests/testdata/wizard-expectations -name pom.xml -type f); 
+              do
+                sed -i -e 's|<mwe2Version>.*</mwe2Version>|<mwe2Version>${env.MWE2_VERSION}</mwe2Version>|' \$f
+              done
+              sed -i -e 's|https://.*/emft/mwe/updates/[^"]*|${env.MWE_P2_URL}|' org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TargetPlatformProject.xtend
```

Comment:
needs rework regaring XtextVersion.java

Reply:
Updated. Found a way to replace the next line after match with sed, which was required to change the getters' return values in XtextVersion

https://ci.eclipse.org/xtext/job/experimental/job/bot-updates-test/20/console

```
23:38:17  diff --git a/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java b/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java
23:38:17  index ac6c8eb..6a430e2 100644
23:38:17  --- a/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java
23:38:17  +++ b/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java
23:38:17  @@ -52,14 +52,14 @@ public class XtextVersion {
23:38:17   	 * @noreference
23:38:17   	 */
23:38:17   	public String getMweVersion() {
23:38:17  -		return "2.12.0.M2";
23:38:17  +		return "2.11.1.M3";
23:38:17   	}
23:38:17   	
23:38:17   	/**
23:38:17   	 * @noreference
23:38:17   	 */
23:38:17   	public String getMweBuildNumber() {
23:38:17  -		return "S202010292115";
23:38:17  +		return "S201910141054";
23:38:17   	}
```